aclogview/CM_Communication.cs

```diff
@@ -176,6 +231,96 @@ public override void contributeToTreeView(TreeView treeView)
         }
     }*/
 
+    public class Emote : Message
+    {
+        public PStringChar emoteMessage;
+
+        public static Emote read(BinaryReader binaryReader)
+        {
+            var newObj = new Emote();
+            newObj.emoteMessage = PStringChar.read(binaryReader);
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("emoteMessage = " + emoteMessage);
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+
+    public class HearEmote : Message
+    {
+        public PStringChar EmoteMessage;
+        public PStringChar SenderName;
+        public uint SenderID;
+
+        public static HearEmote read(BinaryReader binaryReader)
+        {
+            var newObj = new HearEmote();
+            newObj.SenderID = binaryReader.ReadUInt32();
+            newObj.SenderName = PStringChar.read(binaryReader);
+            newObj.EmoteMessage = PStringChar.read(binaryReader);
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("EmoteMessage = " + EmoteMessage.m_buffer);
+            rootNode.Nodes.Add("SenderName = " + SenderName.m_buffer);
+            rootNode.Nodes.Add("SenderID = " + Utility.FormatGuid(this.SenderID));
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+    public class SoulEmote : Message
+    {
+        public PStringChar EmoteMessage;
+
+        public static SoulEmote read(BinaryReader binaryReader)
+        {
+            SoulEmote newObj = new SoulEmote();
+            newObj.EmoteMessage = PStringChar.read(binaryReader);
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("EmoteMessage = " + EmoteMessage.m_buffer);
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+    public class HearSoulEmote : Message
+    {
+        public PStringChar EmoteMessage;
+        public PStringChar SenderName;
+        public uint SenderID;

```

Comment:
Same here - Please reorder the declarations and order of nodes in the tree view to match the read order